server/odc-migrate/src/main/java/com/oceanbase/odc/migrate/jdbc/common/V42417HistoricalPartitionPlanMigrate.java

```diff
@@ -0,0 +1,310 @@
+import com.oceanbase.odc.service.partitionplan.model.PartitionPlanConfig;
+import com.oceanbase.odc.service.partitionplan.model.PartitionPlanStrategy;
+import com.oceanbase.odc.service.partitionplan.model.PartitionPlanTableConfig;
+
+import lombok.Getter;
+import lombok.Setter;
+import lombok.extern.slf4j.Slf4j;
+
+/**
+ * {@link V42417HistoricalPartitionPlanMigrate}
+ *
+ * @author yh263208
+ * @date 2024-03-11 17:07
+ * @since ODC_release_4.2.4
+ */
+@Slf4j
+@Migratable(version = "4.2.4.17", description = "migrate historical partition plan data")
+public class V42417HistoricalPartitionPlanMigrate implements JdbcMigratable {
+
+    @Override
+    public void migrate(DataSource dataSource) {
+        JdbcTemplate jdbcTemplate = new JdbcTemplate(dataSource);
+        List<PartitionPlanEntityWrapper> partitionPlanWrappers = generatePartitionPlanEntities(jdbcTemplate);
+        if (CollectionUtils.isEmpty(partitionPlanWrappers)) {
+            return;
+        }
+        TransactionTemplate transactionTemplate = JdbcOperationsUtil.getTransactionTemplate(dataSource);
+        transactionTemplate.execute(status -> {
+            try {
+                batchCreatePartiPlans(jdbcTemplate, new ArrayList<>(partitionPlanWrappers));
+                partitionPlanWrappers.forEach(i -> i.getPartitionPlanTableEntityWrappers()
+                        .forEach(j -> j.setPartitionPlanId(i.getId())));
+                batchCreatePartiPlanTables(jdbcTemplate, partitionPlanWrappers.stream()
+                        .flatMap(i -> i.getPartitionPlanTableEntityWrappers().stream())
+                        .collect(Collectors.toList()));
+                partitionPlanWrappers.forEach(i -> i.getPartitionPlanTableEntityWrappers()
+                        .forEach(j -> j.getPartitionKeyEntities()
+                                .forEach(k -> k.setPartitionplanTableId(j.getId()))));
+                batchCreatePartiPlanTableKeys(jdbcTemplate, partitionPlanWrappers.stream()
+                        .flatMap(i -> i.getPartitionPlanTableEntityWrappers().stream()
+                                .flatMap(j -> j.getPartitionKeyEntities().stream()))
+                        .collect(Collectors.toList()));
+                batchUpdateScheduleParameterJson(jdbcTemplate, partitionPlanWrappers);
+            } catch (Exception e) {
+                log.warn("Failed to migrate historical partition plan", e);
+                status.setRollbackOnly();
+                throw new IllegalStateException("failed to migrate historical partition plan");
+            }
+            return null;
+        });
+    }
+
+    private List<PartitionPlanEntityWrapper> generatePartitionPlanEntities(JdbcTemplate jdbcTemplate) {
+        List<PartitionPlanEntityWrapper> wrappers = jdbcTemplate.query("select id, "
+                + "flow_instance_id, creator_id, modifier_id, database_id, schedule_id "
+                + "from connection_partition_plan where is_config_enabled=1", (rs, rowNum) -> {
```

Comment:
seems we should also migrate for disabled configuration ?

Reply:
disabled config is useless, so why should we migrate them?